src/classes/CCON_Faculty_TDTM.cls

```diff
@@ -65,169 +68,179 @@ public with sharing class CCON_Faculty_TDTM extends TDTM_Runnable {
-                        potentialAfflDeletionsIds.add(oldEnroll.Affiliation__c);
-                    }
+            potentialAfflDeletions = getAffiliations(potentialAfflDeletionsIds);
 
-                    potentialAfflDeletions = getAffiliations(potentialAfflDeletionsIds);
-
-                    for (Integer i = 0; i < oldList.size(); i++) {
-
-                        Course_Enrollment__c oldEnroll = (Course_Enrollment__c)oldList[i];
-
-                        if (oldEnroll.get('RecordTypeId') == facultyRecType) {
-                            if (potentialAfflDeletions.containsKey(oldEnroll.Affiliation__c)) {
-                                Affiliation__c affl = potentialAfflDeletions.get(oldEnroll.Affiliation__c);
-                                if (affl.Course_Enrollment__r.size() == 0) {
-                                    affiliationsToDelete.add(affl);
-                                }
-                            }
-                            if (oldEnroll.Primary__c) {
-                                // Update Course Offering to remove Primary Faculty
-                                Course_Offering__c cOff = new Course_Offering__c(
-                                    Id = oldEnroll.Course_Offering__c,
-                                    Faculty__c = null
-                                );
-                                courseOfferingsToUpdate.add(cOff);
-                            }
-                        }
+            for (Integer i = 0; i < oldList.size(); i++) {
+
+                Course_Enrollment__c oldEnroll = (Course_Enrollment__c)oldList[i];
+                if (oldEnroll.get('RecordTypeId') != facultyRecType) {
+                    continue;
+                }
+
+                if (potentialAfflDeletions.containsKey(oldEnroll.Affiliation__c)) {
+                    Affiliation__c affl = potentialAfflDeletions.get(oldEnroll.Affiliation__c);
+                    if (affl.Course_Enrollment__r.size() == 0) {
+                        affiliationsToDelete.add(affl);
                     }
                 }
-            }
 
-            if(courseOfferingsToUpdate.size() > 0) {
-                dmlWrapper.objectsToUpdate.addAll((List<SObject>)courseOfferingsToUpdate);
+                if (oldEnroll.Primary__c) {
+                    // Update Course Offering to remove Primary Faculty
+                    Course_Offering__c cOff = new Course_Offering__c(
+                        Id = oldEnroll.Course_Offering__c,
+                        Faculty__c = null
+                    );
+                    courseOfferingsToUpdate.add(cOff);
+                }
             }
+        }
 
-            if(affiliationsToDelete.size() > 0) {
-                dmlWrapper.objectsToDelete.addAll((List<SObject>)affiliationsToDelete);
-            }
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CCON_Faculty_TDTM, true);
```

Comment:
There's quite a bit of formatting changes in this class, but these few lines are the biggest changes (as well as the reentrancy check on line 77)